verifier/trustpolicy/blob.go

```diff
@@ -0,0 +1,133 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trustpolicy
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/notaryproject/notation-go/dir"
+	set "github.com/notaryproject/notation-go/internal/container"
+	"github.com/notaryproject/notation-go/internal/slices"
+)
+
+// BlobDocument represents a trustpolicy.blob.json document
+type BlobDocument struct {
```

Comment:
Do we need to consider how to support a multi-version trust policy? If we have trust policy v2.0, and the struct fields differ from v1.0 because our verifier uses the v1.0 struct directly, how can we ensure that both v1.0 and v2.0 work without causing breaking changes?

Reply:
That's a good point, we already have this issue for OCI trust policy so IMO we should address it outside of this PR.

Reply:
A common model for trust policy is desired here. If we have multiple trust policy versions, all of them can be transformed to the common model internally. I agree that this can be done in a separate PR in the future.

Reply:
Tracking it in https://github.com/notaryproject/notation-go/issues/415